src/Calculator/GraphingSettings.xaml.cpp

```diff
@@ -0,0 +1,220 @@
+#include "pch.h"
+
+#include "GraphingSettings.xaml.h"
+#include <CalcViewModel\Common\AppResourceProvider.cpp>
+
+using namespace Graphing;
+
+using namespace CalculatorApp;
+using namespace CalculatorApp::Controls;
+
+using namespace Platform;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Controls::Primitives;
+using namespace Windows::UI::Xaml::Data;
+using namespace Windows::UI::Xaml::Input;
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Navigation;
+
+GraphingSettings::GraphingSettings()
+{
+    InitializeComponent();
+
+    auto resLoader = AppResourceProvider::GetInstance();
+
+    x_MinHeading->Text = resLoader.GetResourceString(L"x-min");
+    x_MaxHeading->Text = resLoader.GetResourceString(L"x-max");
+    y_MinHeading->Text = resLoader.GetResourceString(L"y-min");
+    y_MaxHeading->Text = resLoader.GetResourceString(L"y-max");
+
+    GridHeading->Text = resLoader.GetResourceString(L"GridHeading");
+    UnitsHeading->Text = resLoader.GetResourceString(L"UnitsHeading");
+
+    Radians->Content = resLoader.GetResourceString(L"TrigModeRadians");
+    Degrees->Content = resLoader.GetResourceString(L"TrigModeDegrees");
+    Gradians->Content = resLoader.GetResourceString(L"TrigModeGradians");
+}
+
+void GraphingSettings::GraphingCalculator_DataContextChanged(FrameworkElement ^ sender, DataContextChangedEventArgs ^ args)
+{
+    // The graph control we are working with
+    m_ParentGrapher = (GraphControl::Grapher ^) args->NewValue;
+}
+
+void GraphingSettings::TrigUnitModeClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    if (Degrees->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Degrees;
+    }
+
+    if (Radians->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Radians;
+    }
+
+    if (Gradians->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Grads;
+    }
+}
+
+void GraphingSettings::UpdateExtents(Platform::Object ^ sender)
+{
+    FrameworkElement ^ fe = (FrameworkElement ^) sender;
+    FrameworkElement ^ currentFocus = (FrameworkElement ^) FocusManager::GetFocusedElement();
+
+    bool thisControlHasFocus = (fe == currentFocus);
```

Comment:
Would fe->FocusState != ::FocusState::Unfocused work? I am not sure of the implementation of FocusManager::GetFocusedElement() but it may be doing additional work to find the focused control when we could just check the FocusState on the control we already have.

Reply:
Looks like it.  Changed.